app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js

```diff
@@ -116,6 +117,51 @@ window.GOVUK.analyticsGa4.analyticsModules = window.GOVUK.analyticsGa4.analytics
       }
     },
 
+    populateLinkPathParts: function (clickData) {
+      var href = clickData.url
+      var path = ''
+      if (this.hrefIsRelative(href)) {
+        path = href
+      } else if (this.isMailToLink(href)) {
+        path = href
+      } else {
+        path = href.replace(/^(http:||https:)?(\/\/)([^\/]*)/, '') // eslint-disable-line no-useless-escape
+      }
+
+      if (path === '/' || path.length === 0) {
+        return
+      }
+
+      /* Divide the length into hundreds, to determine how many string "parts" to make.
+      For example a 150 char string length divided by 100 would return 1.5,
+      so we round up, which returns 2, which makes sense as we would need one segment for chars 0-100, and another for chars 100-150 . */
+      var hundreds = Math.ceil(path.length / 100)
+
+      var LIMIT = 5
+
+      if (hundreds > LIMIT) {
+        hundreds = LIMIT // Limit the amount of data we send to 500 chars
+      }
+
+      clickData.link_path_parts = {}
+
+      for (var i = 0; i < LIMIT; i++) {
+        var stringIndex = (i + 1).toString() // Start the GTM index from 1 for analysts benefit
+        clickData.link_path_parts[stringIndex] = undefined // Ensure the object keys are cleared, so link click paths aren't mixed on separate pushes
+      }
+
+      for (i = 0; i < hundreds; i++) {
```

Comment:
There's probably a reason for this, but would it be possible to combine this loop with the loop above? They seem related. It's also a little confusing why one uses `LIMIT` and the other uses `hundreds`, when you've already compared the two. Why is that?

Reply:
@andysellick I basically need to create the 5 object keys first and set them as `undefined`. It uses LIMIT as that's how many keys we want the object to have.  We need all 5 values to be set to undefined, so that any previous values from a previous push will be overwritten.

Then, I can overwrite the `undefined` with the string values. This uses `hundreds` as it allows us to only loop the amount of times needed to add the string to the object. If we used `LIMIT`, we are always assuming that the string will fill up 5 segments. With `hundreds`, we're getting the amount of segments based on the strings length, and then if it hasn't taken up 5 keys, we are sending `undefined` for the remaining keys which we know exist because of our previous loop.

If I were to combine them, it would get complicated, because I would be trying to add the string to the object, and then if the string has less than 5 segments, I'm then going to need to add the keys with the value `undefined` after it to ensure the object always has 5 keys. 

Essentially it's like how you created the `eventSchema`, we:

- Create the schema object with the keys that will exist in GTM, all of them set as undefined
- Add values to any keys which are relevant

I guess the reason the object is created in a loop is so that the `LIMIT` value can simply be modified, and then the schema object, and the max amount of segments that can be added to the object are both updated accordingly.

Reply:
I think that this whole section could be written more simply. I've just tried replacing 134-159 with this, and it seems to work. Can you check and see if this would be a suitable alternative?

```JavaScript
var parts = path.match(/.{1,100}/g)
var obj = {}
for (var i = 0; i < 5; i++) {
  obj[(i + 1).toString()] = parts[i]
}
clickData.link_path_parts = obj
```

Reply:
Nice solution, thanks @andysellick